### PR TITLE
Adds Stack data structure and testing

### PR DESCRIPTION
**Pull request by primeXXIX**

### CODE REVIEW

**Comment by nicosesma** [comment]
@primeXXIX This PR should be getting merged to your master, not GuildCrafts master

**Comment by primeXXIX** [comment]
how do you change that?

File: spec/stack.js [hunk 1]
```diff
@@ -3,20 +3,55 @@ import chaiChange from 'chai-change'
...
+    'use strict'
+
+    it('exists', () => {
+        expect(Stack).to.be.a('function')
+    })
+
+    context('push()', () => {
+        it('pushes an element to the top of the stack.', () => {
+            const myStack = new Stack()
+            expect(() => myStack.pushie('foo'))
```
Comment (jrobcodes) [inline spec/stack.js:16]: `pushie` is not the name of the function required by the spec.  `push` is *the* name for this behavior in a `Stack` data structure.

File: spec/stack.js [hunk 2]
```diff
@@ -3,20 +3,55 @@ import chaiChange from 'chai-change'
...
+                .to.alter(() => myStack.lengthy(), {
```
Comment (jrobcodes) [inline spec/stack.js:17]: `lengthy` is not the name required by the specification.

File: spec/stack.js [hunk 3]
```diff
@@ -3,20 +3,55 @@ import chaiChange from 'chai-change'
...
-
 describe('Stack', () => {
-  'use strict'
+    'use strict'
+
+    it('exists', () => {
+        expect(Stack).to.be.a('function')
+    })
+
+    context('push()', () => {
```
Comment (jrobcodes) [inline spec/stack.js:13]: An additional test is required in order to determine if the correct value was pushed (the current test only checks to see if the size of the stack changed).

**Review by jrobcodes — COMMENTED** [review]
Please update your API to match the specification defined in the goal, and resubmit your PR.

**Pull request closed without merging**